python/katana/local/_graph.pyx

```diff
@@ -481,6 +531,182 @@ cdef class GraphBase:
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        cdef string input_str = bytes(name, "utf-8")
+        if not manager.HasAtomicType(input_str):
+            return None
+        return manager.GetEntityTypeID(input_str)
+
+    def node_entity_type_to_type_name_set(self, entity_type_id: int):
+        """
+        Return a set of strings representing the names of the types for the type ID
+        `entity_type_id` for nodes
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        return handle_result_TypeNameSet(manager.EntityTypeToTypeNameSet(entity_type_id))
+
+    def edge_entity_type_to_type_name_set(self, entity_type_id: int):
+        """
+        Return a set of strings representing the names of the types for the type ID
+        `entity_type_id` for edges
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        return handle_result_TypeNameSet(manager.EntityTypeToTypeNameSet(entity_type_id))
+
+    def get_atomic_node_type_name(self, entity_type_id: int):
+        """
+        Return the type name if `entity_type_id` is a valid node entity type ID. Otherwise,
+        return None
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        r =  manager.GetAtomicTypeName(entity_type_id)
+        if r:
+            return r.value()
+        return None
+
+    def get_atomic_edge_type_name(self, entity_type_id: int):
+        """
+        Return the type name if `entity_type_id` is a valid edge entity type ID. Otherwise,
+        return None
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        r =  manager.GetAtomicTypeName(entity_type_id)
+        if r:
+            return r.value()
+        return None
+
+    def get_atomic_node_entity_type_ids(self):
+        """
+        Return a list of all node entity type IDs
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        return manager.GetAtomicEntityTypeIDs()
+
+    def get_atomic_edge_entity_type_ids(self):
+        """
+        Return a list of all edge entity type IDs
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        return manager.GetAtomicEntityTypeIDs()
+
+    def node_is_subtype_of(self, sub_type: int, super_type: int):
+        """
+        Return True iff the type `sub_type` is a subtype of the type `super_type` for
+        nodes
+        """
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        return manager.IsSubtypeOf(sub_type, super_type)
+
+    def edge_is_subtype_of(self, sub_type: int, super_type: int):
```

Comment:
This should support int *and* `EntityType` objects. You will need to do a dynamic check with `isinstance`. Python doesn't have overloading since it doesn't have type based dispatch as such.

Reply:
This change should be made to several functions in `EntityTypeManager`